src/components/calcite-popover-manager/calcite-popover-manager.tsx

```diff
@@ -52,9 +52,22 @@ export class CalcitePopoverManager {
   //
   //--------------------------------------------------------------------------
 
-  getRelatedPopover = (element: HTMLElement): HTMLCalcitePopoverElement => {
-    const { selector, el } = this;
-    const id = element.closest(selector)?.getAttribute(POPOVER_REFERENCE);
```

Comment:
closest wasn't working in shadowDom cases. Replacing with composedPath fixes this.